### PR TITLE
use node 14

### LINKED ISSUE #944: Update node version
I noticed this warning, the package node version is set to >=6 and travis is set to 8. We should bring this up to 10 or 12:

> npm WARN notsup Unsupported engine for @mapbox/assembly@0.24.0: wanted: {"node":">=6","npm":"^5.0.0"} (current: {"node":"12.7.0","npm":"6.13.2"})

### PR DESCRIPTION
**Pull request by samanpwbb**
Minimal changes to get us to node 14. 

Closes #944

@tristen can you give this a look?

Thanks!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
     "uglify-js": "^3.4.3"
   },
   "devDependencies": {
-    "@mapbox/batfish": "2.1.0",
+    "@mapbox/batfish": "2.2.0",
```
Comment (samanpwbb) [inline package.json:70]: This is not strictly necessary for the node 14 upgrade but I figured I'd sneak it in.

File: package.json [hunk 2]
```diff
@@ -98,7 +98,7 @@
     "pretty-bytes": "^5.1.0",
     "prop-types": "^15.6.2",
     "react": "^16.4.1",
-    "react-dom": "^16.4.1",
+    "react-dom": "^16.14.0",
```
Comment (samanpwbb) [inline package.json:101]: I ran `npm audit fix`, which produced this change and the lodash change.

**Comment by samanpwbb** [comment]
Looks like we have some build issues to sort out

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 language: node_js
 node_js:
-  - "8"
+  - "14"
 cache:
   directories:
     - $HOME/.npm
 before_install:
-  - npm install npm@5
+  - npm install npm@7
```
Comment (tristen) [inline .travis.yml:8]: Is any of this still necessary?

Comment (samanpwbb) [reply .travis.yml:8]: Looks like travisCI will look at `.nvmrc` files but not at package.json `engines` 👉  https://github.com/travis-ci/travis-ci/issues/4653#issuecomment-169035161

**Comment by samanpwbb** [comment]
@tristen can you give this another look?

**Review by tristen — APPROVED** [review]
🐕‍🦺

**Pull request merged**